module/zfs/btree.c

```diff
@@ -0,0 +1,2110 @@
+/*
+ * CDDL HEADER START
+ *
+ * This file and its contents are supplied under the terms of the
+ * Common Development and Distribution License ("CDDL"), version 1.0.
+ * You may only use this file in accordance with the terms of version
+ * 1.0 of the CDDL.
+ *
+ * A full copy of the text of the CDDL should have accompanied this
+ * source.  A copy of the CDDL is also available via the Internet at
+ * http://www.illumos.org/license/CDDL.
+ *
+ * CDDL HEADER END
+ */
+/*
+ * Copyright (c) 2019 by Delphix. All rights reserved.
+ */
+
+#include	<sys/btree.h>
+#include	<sys/bitops.h>
+#include	<sys/zfs_context.h>
+
+kmem_cache_t *btree_leaf_cache;
+
+/*
+ * Control the extent of the verification that occurs when btree_verify is
+ * called. Primarily used for debugging when extending the btree logic and
+ * functionality.
+ */
+#ifdef ZFS_DEBUG
+int btree_verify_intensity = 5;
+#else
+int btree_verify_intensity = 0;
+#endif
+
+/*
+ * Purely a convenience function, for simpler argument ordering and silencing
+ * compiler warnings about return values. We would use bcopy, but the kernel
+ * version doesn't handle overlapping src and dest.
+ */
+static inline void
+bmov(const void *src, void *dest, size_t size)
+{
+	(void) memmove(dest, src, size);
+}
+
+static void
+btree_poison_node(btree_t *tree, btree_hdr_t *hdr)
```

Comment:
`btree_poison_node()` and `btree_verify_poison_helper()` are only used in debug builds.  Can you wrap them with `#ifdef ZFS_DEBUG` to silence the following warnings for non-debug builds.

```
.../zfs/module/zfs/btree.c:48:1: warning: ‘btree_poison_node’ defined but not used [-Wunused-function]
 btree_poison_node(btree_t *tree, btree_hdr_t *hdr)
 ^
.../zfs/module/zfs/btree.c:2051:1: warning: ‘btree_verify_poison_helper’ defined but not used [-Wunused-function]
 btree_verify_poison_helper(btree_t *tree, btree_hdr_t *hdr)
 ^
```